### PR TITLE
Closes #1066: Add larger space around content options to paragraph types

### LINKED ISSUE #1066: Add larger space around content options
Apply to Text on media and Text with background paragraph types

### PR DESCRIPTION
**Pull request by danahertzberg**
## Description
Add larger space around content options to paragraph types

Review site: https://00505a9a-3ed0-4b8e-ab3f-3c3e6db30fbf.probo.build/

## Related Issue
Closes #1066 
Dependent on https://github.com/az-digital/arizona-bootstrap/issues/375

## How Has This Been Tested?
Probo

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I've added this Pull Request to the AZ Quickstart Github project and set it to "Review in progress".
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by trackleft — DISMISSED** [review]

**Review by mmunro-ltrr — DISMISSED** [review]
Built locally with Lando, copied in the CSS from arizona-bootstrap PR 346, set theme to use the local CSS, enabled demo content, verified that the new settings were available when editing the Text on Media demo.

**Review by joeparsons — APPROVED** [review]

File: modules/custom/az_paragraphs/src/Plugin/paragraphs/Behavior/AZTextWithBackgroundParagraphBehavior.php [hunk 1]
```diff
@@ -93,6 +93,8 @@ public function buildBehaviorForm(ParagraphInterface $paragraph, array &$form, F
         'py-8' => $this->t('8 (6.0rem | ~96px)'),
         'py-9' => $this->t('9 (7.0rem | ~112px)'),
         'py-10' => $this->t('10 (8.0rem | ~128px)'),
+        'py-20' => $this->t('20 (16.0rem | ~256px)'),
+        'py-30' => $this->t('30 (24.0rem | ~384px)'),
```
Comment (danahertzberg) [inline modules/custom/az_paragraphs/src/Plugin/paragraphs/Behavior/AZTextWithBackgroundParagraphBehavior.php:97]:
```suggestion
        'py-md-30 py-10' => $this->t('30 (24.0rem | ~384px)'),
```

File: modules/custom/az_paragraphs/src/Plugin/paragraphs/Behavior/AZTextWithBackgroundParagraphBehavior.php [hunk 2]
```diff
@@ -93,6 +93,8 @@ public function buildBehaviorForm(ParagraphInterface $paragraph, array &$form, F
         'py-8' => $this->t('8 (6.0rem | ~96px)'),
         'py-9' => $this->t('9 (7.0rem | ~112px)'),
         'py-10' => $this->t('10 (8.0rem | ~128px)'),
+        'py-20' => $this->t('20 (16.0rem | ~256px)'),
```
Comment (danahertzberg) [inline modules/custom/az_paragraphs/src/Plugin/paragraphs/Behavior/AZTextWithBackgroundParagraphBehavior.php:96]: Desired classes: `py-10` `py-md-20`

File: modules/custom/az_paragraphs/src/Plugin/paragraphs/Behavior/AZTextWithBackgroundParagraphBehavior.php (hunk 1 above)
Comment (danahertzberg) [inline modules/custom/az_paragraphs/src/Plugin/paragraphs/Behavior/AZTextWithBackgroundParagraphBehavior.php:97]: Desired classes: `py-10` `py-md-30`

File: modules/custom/az_paragraphs/src/Plugin/paragraphs/Behavior/AZTextWithMediaParagraphBehavior.php [hunk 3]
```diff
@@ -135,6 +135,8 @@ public function buildBehaviorForm(ParagraphInterface $paragraph, array &$form, F
         'y-8' => $this->t('8 (6.0rem | ~96px)'),
         'y-9' => $this->t('9 (7.0rem | ~112px)'),
         'y-10' => $this->t('10 (8.0rem | ~128px)'),
+        'y-20' => $this->t('20 (16.0rem | ~256px)'),
```
Comment (danahertzberg) [inline modules/custom/az_paragraphs/src/Plugin/paragraphs/Behavior/AZTextWithMediaParagraphBehavior.php:138]: Desired classes: `py-10` `py-md-20` OR `my-10` `my-md-20`

File: modules/custom/az_paragraphs/src/Plugin/paragraphs/Behavior/AZTextWithMediaParagraphBehavior.php [hunk 4]
```diff
@@ -135,6 +135,8 @@ public function buildBehaviorForm(ParagraphInterface $paragraph, array &$form, F
...
+        'y-30' => $this->t('30 (24.0rem | ~384px)'),
```
Comment (danahertzberg) [inline modules/custom/az_paragraphs/src/Plugin/paragraphs/Behavior/AZTextWithMediaParagraphBehavior.php:139]: Desired classes: `py-10` `py-md-30` OR `my-10` `my-md-30`

**Comment by joeparsons** [comment]
I've updated this PR with some additional preprocess logic in both the Text on Media and Text with Background Paragraph behavior plugins to conditionally add responsive spacing classes based on the rules @danahertzberg specified.  In working on this, I noticed that the spacing options are not used at all when the `bottom` style is selected for Text on media paragraphs so I also updated the form for that behavior plugin to hide the spacing options when the bottom style is selected.

**Comment by danahertzberg** [comment]
Needless to say... I'm _super_ excited about this.
https://00505a9a-3ed0-4b8e-ab3f-3c3e6db30fbf.probo.build/dana

Approved, but I can't approve my own PR lol

**Review by joeparsons — APPROVED** [review]

**Review by smayden — APPROVED** [review]
!!

**Review by camikazegreen — APPROVED** [review]

**Review by mmunro-ltrr — APPROVED** [review]

**Review by trackleft — APPROVED** [review]

**Pull request merged**